### PR TITLE
Create a base page structure to build upon

### PR DESCRIPTION
**Pull request by luizmugnaini**
Adds the basic structure for building the content for the base page via `dioxus` using the [fullstack](https://dioxuslabs.com/learn/0.4/getting_started/fullstack) approach.

To work with the project you can use `cargo just build` and `cargo just run`, which will both build the `dioxus` dependencies and such, and later will use cargo with the build or run commands.

For more information on what is done in this PR the following references can be used:
* [Dioxus docs](https://dioxuslabs.com/learn/0.4/)
* [Dioxus template example](https://github.com/DioxusLabs/dioxus-template/)
* [dxps/fullstack-rust-axum-dioxus-rwa](https://github.com/dxps/fullstack-rust-axum-dioxus-rwa/)

### CODE REVIEW

File: crates/frontend/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+#[derive(Routable, Debug, Clone, PartialEq)]
+enum Route {
+    #[at("/")]
+    Home,
+}
+
+fn switch(route: Route) -> Html {
+    match route {
+        Route::Home => html! { <Home /> },
```
Comment (luizmugnaini) [inline crates/frontend/src/main.rs:20]: Silly question but should we really write `<Home />`? It seems to be the way people tend to write it although I prefer `<Home/>`.

Comment (luizmugnaini) [inline crates/frontend/src/main.rs:20]: Silly question but should we really write `<Home />`? It seems to be the way people tend to write it although I prefer `<Home/>`.

**Comment by isinyaaa** [comment]
As you're dealing with `cargo-make` on #9, you can just drop [3/6] and [4/6]. I propose that you first create the workspaces on the first commit, then squash [5/6] with the second commit in which you introduce the packages and whatnot. [6/6] should also be squashed on the first and second commits accordingly.

**Comment by luizmugnaini** [comment]
I decided to restructure the whole PR. Now it contains only the changes pertaining to the exact topic specified in the PR title. I'll address the other changes in different PRs.

**Comment by luizmugnaini** [comment]
Last push merges #10

**Comment by isinyaaa** [comment]
The structure looks nice now, thanks for the update! I'll review it ASAP (probably later today).

**Comment by luizmugnaini** [comment]
#9 is now merged

**Comment by luizmugnaini** [comment]
The full rewrite from `yew` and `trunk` to `dioxus` is, as far as this PR is concerned, completed.

**Comment by isinyaaa** [comment]
In [3/7], remove "this" from the commit description:

```diff
-This unfortunately stable makes it impossible to use the options
+Unfortunately stable makes it impossible to use the options
```

**Comment by isinyaaa** [comment]
In [4/7], use the imperative (`Add dependencies [...]`)

**Comment by isinyaaa** [comment]
IMO you can just squash [1/7], [2/7], [4/7], [5/7] and [6/7], so that we are moving from trunk to dioxus, AND adding the necessary gitignore stuff, AND making the necessary changes to Cargo.toml + Justfile + Rust aliases at the same time. This should comprise a single change so it shouldn't be necessary to add multiple commits with the same stuff. Finally, you can move [3/7] just before [7/7].

**Review by isinyaaa — APPROVED** [review]
Rebase this MF and we're good to go 🔥

File: Cargo.toml [hunk 2]
```diff
@@ -4,10 +4,15 @@ version = "0.1.0"
...
+dioxus-fullstack = "0.4.1"
+
+[features]
+default = []
+ssr = ["dioxus-fullstack/axum"]
+web = ["dioxus-fullstack/web"]
 
-[package.metadata.bin]
-trunk = { version = "0.17.5", bins = ["trunk"] }
+[workspace.metadata.bin]
```
[the remark this answers is not in the document]
Comment (luizmugnaini) [reply Cargo.toml:25]: Well that's strange. Will fix.

**Pull request merged**